modules/openxr/scene/openxr_composition_layer_cylinder.cpp

```diff
@@ -188,3 +188,49 @@ void OpenXRCompositionLayerCylinder::set_fallback_segments(uint32_t p_fallback_s
 uint32_t OpenXRCompositionLayerCylinder::get_fallback_segments() const {
 	return fallback_segments;
 }
+
+Vector2 OpenXRCompositionLayerCylinder::intersects_ray(const Vector3 &p_origin, const Vector3 &p_direction) const {
+	Vector3 cylinder_origin = get_position();
```

Comment:
Probably need to use `get_global_position` and `get_global_transform`? There is no guarantee the composition layer doesn't have a parent.

Or alternatively, if the code works in local space, you need to localize the inputs:
```
Transform3D inv_transform = global_transform.inverse();
Vector3 ray_origin = inv_transform * p_origin;
Vector3 ray_direction = inv_transform.basis * p_direction;
```